geoutils/georaster/raster.py

```diff
@@ -846,7 +835,24 @@ def data(self, new_data: np.ndarray | np.ma.masked_array) -> None:
                 f"New data must be of the same shape as existing data: {orig_shape}. Given: {new_data.shape[1:]}."
             )
 
-        self._data = np.ma.masked_array(new_data, fill_value=self.nodata)
+        # If the new data is not masked and has non-finite values, we define a fill_value
```

Comment:
Maybe you can update the docstring of data.setter to explain all this?

Reply:
Done